pallets/exchange/src/lib.rs

```diff
@@ -627,7 +633,7 @@ impl<T: Config> Resolver<T::AccountId, Intention<T>, Error<T>> for Pallet<T> {
 				if rest_in_diff.is_none() || rest_out_diff.is_none() {
```

Comment:
by the way, do we need `rest_out_diff.is_none()` check? Seems it unnecessary, because there is already a check `else if amount_a_in < amount_b_out` for this scope